### PR TITLE
[WIP] Adapting ingest pipelines to new date parsing in ES

### PR DESCRIPTION
**Pull request by ycombinator**
Elasticsearch recently changed how it parses timestamps, which introduced a [regression](https://github.com/elastic/elasticsearch/issues/38345). Ideally, the regression can be fixed but this PR is the "backup plan".

This PR tries to adapt Filebeat's ingest pipelines to work with both the old timestamp parsing logic in Elasticsearch (in case users run Filebeat with an older version of ES) as well as the new parsing logic.

As a demonstration, at the moment this PR only adapts the pipelines in the `elasticsearch` Filebeat module.

### CODE REVIEW

File: filebeat/module/elasticsearch/deprecation/test/test-json.log-expected.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [
     {
+        "@timestamp": "2019-01-30T14:16:20.233Z",
```
Comment (ycombinator) [inline filebeat/module/elasticsearch/deprecation/test/test-json.log-expected.json]: Good news: the `@timestamp` field is back.

Bad news: if you compare the value of this field vs. the value in the [raw log line](https://github.com/elastic/beats/blob/1e3b7ef50ffbf1edd63e989d5d110fbf00dc52b8/filebeat/module/elasticsearch/deprecation/test/test-json.log#L1) that generated it, you'll notice that the time zone in this field, `Z`, is incorrect. It should be `-0800` instead. Alternatively, the time here should be `22:16:20.233Z`.

Comment (ycombinator) [reply filebeat/module/elasticsearch/deprecation/test/test-json.log-expected.json]: Same goes for all other `@timestamp` fields in this PR that are being generated from timestamps  **with time zones** in the raw logs.

**Comment by ycombinator** [comment]
@webmat @spinscale @ruflin WIP / discussion PR to demonstrate the challenges with trying to adapt Filebeat ingest pipelines to work with the new timestamp parsing logic in ES.

File: filebeat/module/elasticsearch/deprecation/test/test-json.log-expected.json (hunk 1 above)
Comment (ruflin) [reply filebeat/module/elasticsearch/deprecation/test/test-json.log-expected.json]: We definitively need support back for the time zones.

**Comment by ycombinator** [comment]
This PR is no longer necessary as the regressions in ES have been fixed in ES itself! Closing unmerged.

**Pull request closed without merging**